internal/api/upload_test.go

```diff
@@ -0,0 +1,147 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/opg-sirius-finance-admin/internal/model"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUploadUrlSwitching(t *testing.T) {
+	mockClient := SetUpTest()
+	client, _ := NewApiClient(mockClient, "http://localhost:3000", "")
+
+	// Create a real temporary file to pass to the Upload function
+	tempFile, err := os.CreateTemp("", "testfile")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(tempFile.Name()) // Clean up after the test
+	defer tempFile.Close()
+
+	// Write some content to the temp file
+	content := []byte("fake file content")
+	if _, err := tempFile.Write(content); err != nil {
+		t.Fatal(err)
+	}
+	// Reset the file pointer to the beginning
+	if _, err := tempFile.Seek(0, io.SeekStart); err != nil {
+		t.Fatal(err)
+	}
+
+	// Define test cases
+	testCases := []struct {
```

Comment:
could simplify this to `map[string]string`

Reply:
done